diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/DiktatRuleSetProvider.kt

```diff
@@ -33,24 +33,39 @@ import java.io.File
  * this constant will be used everywhere in the code to mark usage of Diktat ruleset
  */
 const val DIKTAT_RULE_SET_ID = "diktat-ruleset"
+const val DIKTAT_ANALYSIS_CONF = "diktat-analysis.yml"
 
-// by the default it is expected to have diktat-analysis.yml configuration in the root folder where ktlint is run
-// otherwise it will use default configuration where some rules are disabled
-class DiktatRuleSetProvider(private var diktatConfigFile: String = "diktat-analysis.yml") : RuleSetProvider {
+/**
+ * by the default it is expected to have diktat-analysis.yml configuration in the root folder where 'ktlint' is run
+ * otherwise it will use default configuration where some rules are disabled
+ * @param diktatConfigFile - configuration file where all configurations for inspections and rules are stored
+ */
+class DiktatRuleSetProvider(private var diktatConfigFile: String = DIKTAT_ANALYSIS_CONF) : RuleSetProvider {
     @Suppress("LongMethod")
     override fun get(): RuleSet {
         log.debug("Will run $DIKTAT_RULE_SET_ID with $diktatConfigFile" +
                 " (it can be placed to the run directory or the default file from resources will be used)")
         if (!File(diktatConfigFile).exists()) {
             // for some aggregators of static analyzers we need to provide configuration for cli
             // in this case diktat would take the configuration from the direcory where jar file is stored
-            val configPathWithFileName = File(DiktatRuleSetProvider::class.java.protectionDomain.codeSource.location.toURI()).absolutePath
-            val i = configPathWithFileName.lastIndexOf(File.separator)
-            val configPath = if (i > -1) configPathWithFileName.substring(0, i) else configPathWithFileName
-            diktatConfigFile = "${configPath}${File.separator}diktat-analysis.yml"
+            val ruleSetProviderPath =
+                    DiktatRuleSetProvider::class
+                            .java
+                            .protectionDomain
+                            .codeSource
+                            .location
+                            .toURI()
+
+            val configPathWithFileName = File(ruleSetProviderPath).absolutePath
+
+            val indexOfName = configPathWithFileName.lastIndexOf(File.separator)
+            val configPath = if (indexOfName > -1) configPathWithFileName.substring(0, indexOfName) else configPathWithFileName
+            diktatConfigFile = "$configPath${File.separator}$DIKTAT_ANALYSIS_CONF"
 
             if (!File(diktatConfigFile).exists()) {
-                log.warn("Configuration file $diktatConfigFile not found in file system, the file included in jar will be used. " +
+                log.warn("Configuration file $DIKTAT_ANALYSIS_CONF not found in directory where diktat is run " +
```

Comment:
File name can be different if user overrides it in a plugin configuration, we should take it into account here too.

Reply:
I can print both names for plugin and for cli

Reply:
I mean, when `DiktatRuleSetProvider` is instantiated directly (now it's like this in maven plugin), `diktatConfigFile` constructor property can have a different value. Yes, I think it'll be correct to print both possible locations, if neither can be opened.

Reply:
fixed